rust/gui-client/src-tauri/src/ipc/linux.rs

```diff
@@ -41,19 +41,22 @@ pub async fn connect_to_service(id: ServiceId) -> Result<ClientStream> {
 
 impl Server {
     /// Platform-specific setup
-    pub(crate) async fn new(id: ServiceId) -> Result<Self> {
+    pub(crate) fn new(id: ServiceId) -> Result<Self> {
         let sock_path = ipc_path(id);
+
+        tracing::debug!(socket = %sock_path.display(), "Creating new IPC server");
+
         // Remove the socket if a previous run left it there
-        tokio::fs::remove_file(&sock_path).await.ok();
```

Comment:
I changed these to blocking functions because the `tokio` ones were suddenly failing in CI. It shouldn't really matter because removing files, creating dirs and setting permissions should be really fast.

Reply:
This is all in the service so can't really block the UI thread.